crates/services/tx_status_manager/src/update_sender.rs

```diff
@@ -256,18 +257,18 @@ impl UpdateSender {
         // left for a given tx_id.
         let mut empty = false;
 
-        if let Some(senders) = senders.get_mut(update.tx_id()) {
+        if let Some(senders) = senders.get_mut(&tx_id) {
             // Retain only senders that are able to receive the update.
-            senders
-                .retain_mut(|sender| sender.try_send(update.clone().into_msg()).is_ok());
```

Comment:
nit: this is a SRP violation. In general I try to avoid operations that couple two different ideas together (in this case sending the message and removing dead senders). Code golf should stay out of production code. Could we move this to a `for` loop perhaps? Then explicitly say "if this fails, it's dead and should be removed".

Reply:
We need to rework this logic I think, but that code was before me=D